internal/test/envtest/environment.go

```diff
@@ -254,11 +257,27 @@ func newEnvironment(uncachedObjs ...client.Object) *Environment {
 		}
 	}
 
+	req, _ := labels.NewRequirement(clusterv1.ClusterNameLabel, selection.Exists, nil)
+	clusterSecretCacheSelector := labels.NewSelector().Add(*req)
+	syncPeriod := 10 * time.Minute
+
 	options := manager.Options{
 		Scheme: scheme.Scheme,
 		Metrics: metricsserver.Options{
 			BindAddress: "0",
 		},
+		Cache: cache.Options{
+			// Namespaces: watchNamespaces,
+			SyncPeriod: &syncPeriod,
+			ByObject: map[client.Object]cache.ByObject{
+				// Note: Only Secrets with the cluster name label are cached.
+				// The default client of the manager won't use the cache for secrets at all (see Client.Cache.DisableFor).
+				// The cached secrets will only be used by the secretCachingClient we create below.
+				&corev1.Secret{}: {
+					Label: clusterSecretCacheSelector,
+				},
+			},
+		},
```

Comment:
I copied this from https://github.com/kubernetes-sigs/cluster-api/commit/ae9464dedd36ebf21e803c41c063a7a874cd4de0#diff-8bdee7925f4d1adc50383754391472636156a2a857a7c76290f6073c8d83f5cd
Not sure if it makes sense

Reply:
seems ok, cc @sbueringer for a double check